documentation/rosetta-modelling-component.md

```diff
@@ -662,6 +663,66 @@ E.g.:
 [DayOfWeekEnum->SAT, DayOfWeekEnum->SUN]
 ```
 
+### Data Type and Record Constructor
+
+Data types and records can be constructed using a syntax similar to JSON, i.e.,
+```
+<type name> {
+  <field name 1>: <field value 1>,
+  <field name 2>: <field value 2>,
+  etc
+}
+```
+
+As an example, consider the following data type representing an employee of a company.
+``` Haskell
+type Employee:
+  name string (1..1)
+  age int (0..1)
+  favouriteAnimal string (0..1)
+  mentor Employee (0..1)
+```
+To construct an employee called "Dwight Schrute" with a mentor called "Michael Scott" within Rosetta, the following syntax can be used.
+``` Haskell
+Employee {
+  name: "Dwight Schrute",
+  age: 42,
+  favouriteAnimal: "bear",
+  mentor: Employee {
+    name: "Michael Scott",
+    age: empty,
+    favouriteAnimal: empty,
+    mentor: empty
+  }
+}
+```
+
```

Comment:
I would add the alternative way of creating the "Dwight Schrute" record, which is first to set "Michael Scott" as an alias, and then use that alias when constructing Dwight.

I also understand that you can use Rosetta expressions within the instantiation syntax, for instance a path extraction, so you could add an example accordingly.

Those 2 points are described in the issue.